### PR TITLE
Minor improvements

### PR DESCRIPTION
**Pull request by klaidliadon**
- Update to latest authcontrol (v.0.0.7)
- The argument middleware.Options is not a pointer anymore
- Added `Options.ApplyDefaults` method
- Switched from logger to slog

### CODE REVIEW

File: cache.go [hunk 1]
```diff
@@ -51,25 +51,24 @@ var _ QuotaCache = (*RedisCache)(nil)
...
 		Host:      cfg.Host,
 		Port:      cfg.Port,
 		MaxIdle:   cfg.MaxIdle,
 		MaxActive: cfg.MaxActive,
 		DBIndex:   cfg.DBIndex,
 		OnError: func(err error) {
-			if logger != nil {
-				logger.Error("redis counter error", slog.Any("error", err))
-			}
+			log.With(slog.Any("error", err)).Error("redis counter error")
```
Comment (david-littlefarmer) [inline cache.go:68]: `slog.With()` is creating a new copy of `*slog.Logger` and returns it, but the returned values is not used here, it just logs the error and then throw away the newly created logger, this doesn't make sense, just log it right away. Revert these changes everywhere please.
```suggestion
			log.Error("redis counter error", slog.Any("error", err))
```

File: cache.go [hunk 2]
```diff
@@ -51,25 +51,24 @@ var _ QuotaCache = (*RedisCache)(nil)
...
 		},
 		OnFallbackChange: func(fallback bool) {
-			if logger != nil {
-				logger.Warn("redis counter fallback", slog.Bool("fallback", fallback))
-			}
+			log.With(slog.Bool("fallback", fallback)).Warn("redis counter fallback")
```
Comment (david-littlefarmer) [inline cache.go:71]:
```suggestion
			log.Warn("redis counter fallback", slog.Bool("fallback", fallback))
```

File: handler_test.go [hunk 3]
```diff
@@ -213,9 +214,9 @@ func TestMiddlewareUseAccessKey(t *testing.T) {
 
 		ctx := middleware.WithTime(context.Background(), now)
 
-		for i, max := 0, cfg.RateLimiter.PublicRate*2; i < max; i += _credits {
+		for i, max := 0, cfg.RateLimiter.PublicRPM*2; i < max; i += _credits {
```
Comment (david-littlefarmer) [inline handler_test.go:217]:
```suggestion
		for i := 0; i < cfg.RateLimiter.PublicRPM*2; i += _credits {
```

File: proto/proto.go [hunk 4]
```diff
@@ -9,20 +7,23 @@ import (
 	"github.com/0xsequence/authcontrol/proto"
 )
 
+//go:generate go run github.com/webrpc/webrpc/cmd/webrpc-gen -schema=proto.ridl -target=golang@v0.15.0 -pkg=proto -server -client -out=./proto.gen.go
+//go:generate go run github.com/webrpc/webrpc/cmd/webrpc-gen -schema=proto.ridl -target=typescript@v0.12.0 -client -out=./clients/quotacontrol.gen.ts
```
Comment (david-littlefarmer) [inline proto/proto.go:11]: return these to top of the file please, and i think we don't need to generate server so we don't need `clients` folder.
```suggestion
//go:generate go run github.com/webrpc/webrpc/cmd/webrpc-gen -schema=proto.ridl -target=golang@v0.15.0 -pkg=proto -client -out=./proto.gen.go
//go:generate go run github.com/webrpc/webrpc/cmd/webrpc-gen -schema=proto.ridl -target=typescript@v0.12.0 -client -out=./quotacontrol.gen.ts
```

**Review by david-littlefarmer — COMMENTED** [review]
Few comments, please revert the logging changes.

File: proto/proto.go (hunk 4 above)
Comment (klaidliadon) [reply proto/proto.go:11]: We need the server of quota control actually

Comment (david-littlefarmer) [reply proto/proto.go]: Aha, ok. Then sorry. Keep it as it was.

File: cache.go [hunk 5]
```diff
@@ -51,7 +50,7 @@ var _ QuotaCache = (*RedisCache)(nil)
 var _ QuotaCache = (*LRU)(nil)
 var _ UsageCache = (*RedisCache)(nil)
 
-func NewLimitCounter(cfg RedisConfig, logger logger.Logger) httprate.LimitCounter {
+func NewLimitCounter(cfg RedisConfig, log *slog.Logger) httprate.LimitCounter {
```
Comment (pkieltyka) [inline cache.go:53]: i think its too soon for this unfortunately because not all of our services use slog yet, but goware/logger supports both slog and zerolog. One day we will do this.. but we need to get everything to slog first.

File: middleware/middleware_ratelimit.go [hunk 6]
```diff
@@ -30,47 +28,43 @@ const (
...
 
+// RateLimitConfig is the configuration for the rate limiter middleware.
 type RateLimitConfig struct {
-	Enabled     bool `toml:"enabled"`
-	PublicRate  int  `toml:"public_requests_per_minute"`
-	AccountRate int  `toml:"user_requests_per_minute"`
-	ServiceRate int  `toml:"service_requests_per_minute"`
+	// Enabled turns on/off the middleware.
+	Enabled bool `toml:"enabled"`
+	// PublicRPM is the rate limit for Public sessions expressed as number of requests per minute.
```
Comment (pkieltyka) [inline middleware/middleware_ratelimit.go:35]: I think you forgot to add `BaseRequestCost` config param..?

Comment (klaidliadon) [reply middleware/middleware_ratelimit.go:35]: BaseRequestCost has been moved to Options:

https://github.com/0xsequence/quotacontrol/blob/96e378ebb5fcb96a471598751eca7dc158547db6/middleware/common.go#L18-L22

Since it was used in two places:

https://github.com/0xsequence/quotacontrol/blob/96e378ebb5fcb96a471598751eca7dc158547db6/middleware/middleware_cost.go#L18-L24

https://github.com/0xsequence/quotacontrol/blob/96e378ebb5fcb96a471598751eca7dc158547db6/middleware/middleware_ratelimit.go#L105-L116

So it is was easy to pass it to both middelware in a clean way, plus we can set the default as we do for other things in here.
